src/assisted_test_infra/test_infra/helper_classes/nodes.py

```diff
@@ -228,3 +229,24 @@ def _are_nodes_network_prepared(self):
 
     def wait_till_nodes_are_ready(self, network_name: str = None):
         self.controller.wait_till_nodes_are_ready(network_name)
+
+    def wait_till_nodes_are_ssh_ready(self):
+        log.info("Wait till %s nodes will be ready for SSH connection", len(self.nodes))
+
+        def _ssh_connection():
+            try:
+                for the_node in self.nodes:
+                    if the_node.ssh_connection is not None:
+                        pass
+            except (TimeoutError, SCPException):
+                return False
+            else:
+                return True
+
+        waiting.wait(
+            lambda: _ssh_connection(),
+            timeout_seconds=180,
+            sleep_seconds=20,
+            waiting_for="Nodes to allow ssh connection",
```

Comment:
few fixes to make the code more clear:
```suggestion
        def _all_nodes_allow_ssh_connection():
            try:
                for node in self.nodes:
                    if node.ssh_connection is None:
                        return False
            except (TimeoutError, SCPException):
                return False

            return True

        waiting.wait(
            lambda: _all_nodes_allow_ssh_connection(),
            timeout_seconds=180,
            sleep_seconds=20,
            waiting_for="nodes to allow ssh connection",
```

Reply:
> few fixes to make the code more clear:

Ack no problem .

Reply:
hmm, you still didn't apply all those changes. please look carefully